org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/UnresolvedElementsSubProcessor.java

```diff
@@ -0,0 +1,1798 @@
+				if (inv.getExpression() == null) {
+					addQualifierToOuterProposal(context, inv, methodBinding, proposals);
+				}
+			}
+			return;
+		}
+
+		if (nDiffs == 1) { // one argument mismatching: try to fix
+			int idx= indexOfDiff[0];
+			Expression nodeToCast= arguments.get(idx);
+			ITypeBinding castType= paramTypes[idx];
+			castType= Bindings.normalizeTypeBinding(castType);
+			if (castType.isWildcardType()) {
+				castType= ASTResolving.normalizeWildcardType(castType, false, nodeToCast.getAST());
+			}
+			if (castType != null) {
+				ITypeBinding binding= nodeToCast.resolveTypeBinding();
+				ITypeBinding castFixType= null;
+				if (binding == null || castType.isCastCompatible(binding)) {
+					castFixType= castType;
+				} else if (JavaModelUtil.is50OrHigher(cu.getJavaProject())) {
+					ITypeBinding boxUnboxedTypeBinding= TypeMismatchSubProcessor.boxUnboxPrimitives(castType, binding, nodeToCast.getAST());
+					if (boxUnboxedTypeBinding != castType && boxUnboxedTypeBinding.isCastCompatible(binding)) {
+						castFixType= boxUnboxedTypeBinding;
+					}
+				}
+				if (castFixType != null) {
+					ASTRewriteCorrectionProposal proposal= TypeMismatchSubProcessor.createCastProposal(context, castFixType, nodeToCast, IProposalRelevance.CAST_ARGUMENT_1);
+					String castTypeName= BindingLabelProvider.getBindingLabel(castFixType, JavaElementLabels.ALL_DEFAULT);
+					String[] arg= new String[] { getArgumentName(arguments, idx), castTypeName};
+					proposal.setDisplayName(Messages.format(CorrectionMessages.UnresolvedElementsSubProcessor_addargumentcast_description, arg));
+					proposals.add(proposal);
+				}
+
+				TypeMismatchSubProcessor.addChangeSenderTypeProposals(context, nodeToCast, castType, false, IProposalRelevance.CAST_ARGUMENT_2, proposals);
+			}
+		}
+
+		if (nDiffs == 2) { // try to swap
+			int idx1= indexOfDiff[0];
+			int idx2= indexOfDiff[1];
+			boolean canSwap= canAssign(argTypes[idx1], paramTypes[idx2]) && canAssign(argTypes[idx2], paramTypes[idx1]);
+			if (canSwap) {
+				Expression arg1= arguments.get(idx1);
+				Expression arg2= arguments.get(idx2);
+
+				ASTRewrite rewrite= ASTRewrite.create(astRoot.getAST());
+				rewrite.replace(arg1, rewrite.createCopyTarget(arg2), null);
+				rewrite.replace(arg2, rewrite.createCopyTarget(arg1), null);
+				{
+					String[] arg= new String[] { getArgumentName(arguments, idx1), getArgumentName(arguments, idx2) };
+					String label= Messages.format(CorrectionMessages.UnresolvedElementsSubProcessor_swaparguments_description, arg);
+					ASTRewriteCorrectionProposal proposal = new ASTRewriteCorrectionProposal(label,
+							context.getCompilationUnit(), rewrite, IProposalRelevance.SWAP_ARGUMENTS);
+					proposals.add(proposal);
+				}
+
+				if (declaringTypeDecl.isFromSource()) {
+					ICompilationUnit targetCU= ASTResolving.findCompilationUnitForBinding(cu, astRoot, declaringTypeDecl);
+					if (targetCU != null) {
+						ChangeDescription[]
```

Comment:
I guess if it's copied code, then we might as well keep the typo, in the hope the above class gets refactored to jdt.core